### PR TITLE
#552 Molecules - FMDatePicker

### PR DESCRIPTION
**Pull request by bryantobing12**
### Detailed description of changes

- Adding `FMDatePicker` component
- Adding `FMDatePicker` story
- Adding `FMDatePicker` test

### Recommendations on how to test the changes

`yarn storybook`

- Check if the component matches the design

### Checklist for developer

- [x] The code builds clean without any errors or warnings
- [x] The code does not contain commented out code
- [x] The code does not log anything to console
- [x] I have added unit test(s) to cover new code and successfully executed ran it
- [x] I have thoroughly tested the new code and any adjacent features it may affect

### Link to related issue(s)

https://app.zenhub.com/workspaces/flick-614155581501e1000f356a5d/issues/hzn-one/tms/552

#### Before

N / A

#### After

https://user-images.githubusercontent.com/46083126/159405776-6f2fb5e6-e77c-479b-97cf-74c4563ceeca.mp4

### CODE REVIEW

File: src/Components/Molecules/FMDatePicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export const FMDatePicker = (props: IFMDatePickerProps) => {
+  const { datePickerType, testID, textFieldProps } = props;
+
+  const renderInput = (params: TextFieldProps) => (
+    <TextField {...params} {...textFieldProps} />
+  );
+
+  return (
+    <Box data-testid={testID} color="primary.main">
+      <LocalizationProvider dateAdapter={AdapterDateFns}>
```
Comment (gilangdarmawan) [inline src/Components/Molecules/FMDatePicker/index.tsx:22]: How about moving `<LocalizationProvider />` to ThemeWrapperProvider?

File: src/Components/Molecules/index.tsx [hunk 2]
```diff
@@ -12,3 +12,4 @@ export * from "./FMTabsMenu";
 export * from "./FMTimeline";
 export * from "./FMViewPicture";
 export * from "./FMDateRangePicker";
+export * from "./FMDatePicker";
```
Comment (gilangdarmawan) [inline src/Components/Molecules/index.tsx:15]: better sort it alphabetically

File: test/Components/Molecules/FMDatePicker/FMDatePicker.test.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import { FMDatePicker } from "@Molecules";
+
+describe("FMDatePicker", () => {
+  it("should match snapshot", () => {
+    const { container } = render(
+      <FMDatePicker
+        testID="date-picker-test"
+        datePickerType="date"
+        value={new Date()}
```
Comment (gilangdarmawan) [inline test/Components/Molecules/FMDatePicker/FMDatePicker.test.tsx:17]: The date must be static so it will match to snapshot everytime

**Review by gilangdarmawan — CHANGES_REQUESTED** [review]

**Comment by bryantobing12** [comment]
@gilangdarmawan updated. Kindly check again. Thanks

**Review by gilangdarmawan — APPROVED** [review]

**Pull request merged**